tests/spread/extensions/ros-meta-hello/task.yaml

```diff
@@ -0,0 +1,68 @@
+summary: Build and run a basic ROS snap using meta-ros extension
+
+kill-timeout: 180m
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+
+  SNAP: catkin-noetic-hello
+  SNAP_DIR: "../../plugins/v2/snaps/$SNAP"
+
+  INTERFACE: ros-noetic
+
+  META_SNAP/catkin_noetic_ros_core: ros-noetic-ros-core
+  EXTENSION/catkin_noetic_ros_core: ros1-noetic-ros-core
+
+  META_SNAP/catkin_noetic_ros_base: ros-noetic-ros-base
+  EXTENSION/catkin_noetic_ros_base: ros1-noetic-ros-base
+
+  META_SNAP/catkin_noetic_robot: ros-noetic-robot
+  EXTENSION/catkin_noetic_robot: ros1-noetic-robot
+
+  META_SNAP/catkin_noetic_desktop: ros-noetic-desktop
+  EXTENSION/catkin_noetic_desktop: ros1-noetic-desktop
+
+# The content snap required for the test to succeed is only
+# available on a subset of all the architectures this testbed
+# can run on.
+systems:
+  - ubuntu-20.04
+  - ubuntu-20.04-64
+  - ubuntu-20.04-amd64
+  - ubuntu-20.04-arm64
+
+prepare: |
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+  # Overwrite the extension to test them all out of a single snap
+  sed -i "s|ros1-noetic|${EXTENSION}|" "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
```

Comment:
Since tests are run in destructive mode, you might want to clear the environment. Especially with all the dependencies subtilities that you did.
Using [create_dpkg_restore_point](https://github.com/snapcore/snapcraft/blob/main/tests/spread/plugins/craft-parts/build-and-run-hello/task.yaml#L19) and [dpkg_restore_point](https://github.com/snapcore/snapcraft/blob/main/tests/spread/plugins/craft-parts/build-and-run-hello/task.yaml#L43).

Additionally, you might want to extend that to snaps to make sure you properly removed the content sharing snaps etc